tests/ignite/metrics/test_epoch_metric.py

```diff
@@ -77,26 +78,28 @@ def compute_fn(y_preds, y_targets):
     output2 = (torch.rand(4, 1), torch.randint(0, 2, size=(4, 1), dtype=torch.long))
     em.update(output2)
 
-    assert all([t.device.type == "cpu" for t in em._predictions + em._targets])
-    assert torch.equal(em._predictions[0], output1[0][:, 0])
-    assert torch.equal(em._predictions[1], output2[0][:, 0])
-    assert torch.equal(em._targets[0], output1[1][:, 0])
-    assert torch.equal(em._targets[1], output2[1][:, 0])
+    assert all([t.device.type == available_device for t in em._predictions + em._targets])
+    assert torch.equal(em._predictions[0].cpu(), output1[0][:, 0].cpu())
+    assert torch.equal(em._predictions[1].cpu(), output2[0][:, 0].cpu())
+    assert torch.equal(em._targets[0].cpu(), output1[1][:, 0].cpu())
+    assert torch.equal(em._targets[1].cpu(), output2[1][:, 0].cpu())
     assert em.compute() == 0.0
 
 
-def test_mse_epoch_metric():
+def test_mse_epoch_metric(available_device):
     def compute_fn(y_preds, y_targets):
         return torch.mean(((y_preds - y_targets.type_as(y_preds)) ** 2)).item()
 
-    em = EpochMetric(compute_fn)
+    em = EpochMetric(compute_fn, device=available_device)
+    assert em._device == torch.device(available_device)
+    dtype = torch.int32 if available_device == "mps" else torch.long
```

Comment:
Is it really necessary? You do not do that in other tests like test_epoch_metric etc. Seems like It is not a hard requirement by MPS